src/sparkle/writer/kafka_writer.py

```diff
@@ -109,3 +109,91 @@ def write(self, df: DataFrame) -> None:
+        unique_identifier_column_name (str): Column name used as the Kafka key.
+        spark (SparkSession): Spark session instance to use.
+    """
+
+    def __init__(
+        self,
+        kafka_options: dict[str, Any],
+        kafka_topic: str,
+        unique_identifier_column_name: str,
+        spark: SparkSession,
+    ) -> None:
+        """Initialize the KafkaBatchPublisher object.
+
+        Args:
+            kafka_options (dict[str, Any]): Kafka options for the connection.
+            kafka_topic (str): The target Kafka topic for writing data.
+            unique_identifier_column_name (str): Column name to be used as Kafka key.
+            spark (SparkSession): The Spark session to be used for writing data.
+        """
+        super().__init__(spark)
+        self.kafka_options = kafka_options
+        self.kafka_topic = kafka_topic
+        self.unique_identifier_column_name = unique_identifier_column_name
+
+    @classmethod
+    def with_config(cls, config: Config, spark: SparkSession, **kwargs: Any) -> "KafkaBatchPublisher":
+        """Create a KafkaBatchPublisher object with a configuration.
+
+        Args:
+            config (Config): Configuration object containing settings for the writer.
+            spark (SparkSession): The Spark session to be used for writing data.
+            **kwargs (Any): Additional keyword arguments.
+
+        Returns:
+            KafkaBatchPublisher: An instance configured with the provided settings.
+        """
+        if not config.kafka_output:
+            raise ValueError("Kafka output configuration is missing")
+
+        c = config.kafka_output
+
+        return cls(
+            kafka_options=c.kafka_config.spark_kafka_config,
+            kafka_topic=c.kafka_topic,
+            unique_identifier_column_name=c.unique_identifier_column_name,
+            spark=spark,
+        )
+
+    def write(self, df: DataFrame) -> None:
+        """Write DataFrame to Kafka by converting it to JSON using the configured primary key.
+
+        This method transforms the DataFrame using the unique identifier column name
+        and writes it to the configured Kafka topic.
+
+        Args:
+            df (DataFrame): The DataFrame to be written.
+
+        Raises:
+            KeyError: If the DataFrame does not have the required 'key' and 'value' columns.
+        """
+        # Convert the DataFrame to a Kafka-friendly format
+        kafka_df = to_kafka_dataframe(self.unique_identifier_column_name, df)
+
+        if "key" not in kafka_df.columns or "value" not in kafka_df.columns:
+            raise KeyError(
+                "The DataFrame must contain 'key' and 'value' columns. "
+                "Ensure that `to_kafka_dataframe` transformation is correctly applied."
+            )
+
+        # fmt: off
+        (
+            kafka_df.write.format("kafka")
+            .options(**self.kafka_options)
+            .option("topic", self.kafka_topic)
```

Comment:
Does this mean we need to initialize one instance of this writer everytime we want to write to a topic? I think it's more flexible to get the topic name when we want to invoke `write`, that way, we can also get multiple topics, and just iterate if needed.

Reply:
This is the same in the other writer I copied from, and the other observation I had. I think I spoke to you about this on slack.

I said something like "the topic is set during the istantiation, so for each different topic we gotta create a different instance of the writer class?"

I kept it like this in the batch writer because it's not my design and IDK where we're going yet. I don't like to change other's designs until I know exactly what the route is.